src/CalculatorUITestFramework/MemoryPanel.cs

```diff
@@ -38,5 +54,110 @@ public ReadOnlyCollection<AppiumWebElement> GetAllMemoryListViewItems()
             OpenMemoryPanel();
             return this.MemoryListView.FindElementsByClassName("ListViewItem");
         }
+
+        /// <summary>
+        /// Opens the Memory Panel and clicks the delete button if it is visible
+        /// </summary>
+        public void ClearMemoryPanel()
+        {
+            this.MemoryLabel.Click();
+
+            try
+            {
+                string sourceC = this.session.PageSource;
+                if (sourceC.Contains("ClearMemoryButton"))
+                {
+                    this.ClearMemory.Click();
+                }
+                else
+                {
+                    return;
+                }
+            }
+            catch (WebDriverException ex)
+            {
+                if (ex.Message.Contains("element could not be located"))
```

Comment:
>ex.Message.Contains("element could not be located") [](start = 20, length = 51)

It looks like we are doing this in a number of other places, but it's generally better practice to key off of the type of exception rather than the message. 
ie:
try
{
...
}
catch(ElementNotFoundException ex)
{
//run code for this exception type
}
catch(OtherTypeOfException ex)
{
//run code for OtherTypeOfException
}

anything that isn't one of the above 2 exception types will just throw.  If an exception is both of those types (ie: one inherits from other) it will only run the first exception handler.